### ISSUE TITLE
RadList with LoadOnDemand feature shows blank after switch the items to a short source.

### ISSUE THREAD

**Issue by gogoout** [telerik/nativescript-ui-feedback#674]
> Please take a minute to read our [NativeScript Code of Conduct](https://github.com/NativeScript/codeofconduct) before proceeding with posting issues or discussing. The purpose of this guide is to make communication and cooperation within our forums a pleasure for you and the other members.

### Did you verify this is a real problem by searching the [NativeScript Forum](https://discourse.nativescript.org/)?
Yes
### Tell us about the problem

RadList with LoadOnDemand feature shows blank after switch the items to a short source.

### Which platform(s) does your issue occur on?
iOS(haven't tested on Android)

### Please provide the following version numbers that your issue occurs with:
- **Progress NativeScript UI version:** (the version of nativescript-telerik-ui/-pro in the `package.json`)
`nativescript-ui-listview@3.5.5`

- **CLI:** (run `tns --version` to fetch it)
`tns --version` `4.0.0`

- **Cross-platform modules:** (the version of tns-core-modules in the `package.json`)
`tns-core-modules@4.0.0` 

- **Runtime(s):** (the `"tns-android"` and `"tns-ios"` properties in the `package.json`)
`tns-ios` `4.0.1`

### Please tell us how to recreate the issue in as much detail as possible.
1. Start the application ..
2. Scroll down to trigger the loadOnDemand() and it works well. (The Console will print `load on demand reset to auto, total:90, current:30`)
3. Scroll up to trigger the pullToRefresh and it works well. (The Console will print `load on demand reset to auto, total:90, current:30`)
4. Click the button "Short" and the RadList will render the short list. (The Console will print `load on demand set to none, total:5, current:5`)
5_1. Click the button "Long" again the RadList will render a blank screen.(The Console will print `load on demand reset to auto, total:90, current:30`)
5_2. Scroll up to trigger the pullToRefresh and  the RadList will render a blank screen

The bug might relate to https://github.com/telerik/nativescript-ui-feedback/issues/595 . Because I have to set LoadOnDemand Mode to none to disabled the loadOnDemand function.

### Is there code involved? If so, please share the minimal amount of code needed to recreate the problem.
_(You can paste entire code snippets or attach a runnable project)_

https://play.nativescript.org/?template=play-ng&id=LpyRpQ

**Comment by gogoout** [comment]
Is there any progress on this bug or is there any work round? The real scenario that I ran into is a list that implement the pullToRefresh and LoadOnDemand at the same time. If the list comes with just a few data and if you perform a pullToRefresh action after that, the list will just go blank.

**Comment by VladimirAmiorkov** [comment]
Hi @gogoout ,

We have recently released a new **3.6.0** version that resolves a couple of issues with "load on demand" including [this](https://github.com/telerik/nativescript-ui-feedback/issues/595) one. Full release notes can be found [here](https://github.com/telerik/nativescript-ui-feedback/blob/master/releases/listview.md#v360).

I tested the attached playground project by downloading it and changing the way you are disabling the "load on demand" with the new functionality of `returnValue` introduced with 3.6.0. In order to disable the "load on demand" simply set the `returnValue` of the arguments in the `loadMoreDataRequested` event to false:

```
_onLoadMore(args: ListViewEventData) {
    const listView: RadListView = args.object;
    if (this.totalCount > -1 && this.totalCount > this.items.length) {
        args.returnValue = true;
        this.onLoadMore.emit({ offset: this.items.length, event: args });
    }
    else {
        console.log('load on demand set to none');
        args.returnValue = false;
    }
}
```

Closing this issue.

**Comment by gogoout** [comment]
Hi there,
I've tested v3.6.0 with ios v4.2.0. However, I believe the issue is not solved.
I replaced all the code of disabling the "load on demand"  , with the code below.
```js

    _onLoadMore(args: LoadOnDemandListViewEventData) {
        const listView: RadListView = args.object;
        if (this._totalCount > -1 && this._totalCount > this._list.length) {
            console.log('on load more');
            args.returnValue = true;
            this.onLoadMore.emit({offset: this._list.length, event: args});
        }
        else {
            console.log('disable load more');
            // listView.notifyLoadOnDemandFinished();
            args.returnValue = false;
        }
    }
```

And result is like this:
![screenshot1](https://user-images.githubusercontent.com/7570175/44376528-12b95200-a54d-11e8-9cb6-d6da67ae1d5c.gif)

As you can see, the activityIndicator is always there and the console log like this:
```cli
CONSOLE LOG file:///app/11.js:752:24: disable load more
CONSOLE LOG file:///app/11.js:752:24: disable load more
CONSOLE LOG file:///app/11.js:752:24: disable load more
CONSOLE LOG file:///app/11.js:752:24: disable load more
CONSOLE LOG file:///app/11.js:752:24: disable load more
CONSOLE LOG file:///app/11.js:752:24: disable load more
CONSOLE LOG file:///app/11.js:752:24: disable load more
CONSOLE LOG file:///app/11.js:752:24: disable load more
CONSOLE LOG file:///app/11.js:752:24: disable load more
CONSOLE LOG file:///app/11.js:752:24: disable load more
CONSOLE LOG file:///app/11.js:752:24: disable load more
CONSOLE LOG file:///app/11.js:752:24: disable load more
CONSOLE LOG file:///app/11.js:752:24: disable load more
CONSOLE LOG file:///app/11.js:752:24: disable load more
CONSOLE LOG file:///app/vendor.js:137971:31: start waiting
CONSOLE LOG file:///app/bundle.js:6615:24: action:  [orders]load orders
CONSOLE LOG file:///app/bundle.js:6619:32: payload:
{"status":"shipped","forceReload":true,"offset":0}
```
All the log "disable load more" is printed when I do the scrolldown action which means the loadOnDemand function is still called multiple times before I perform the PullToRefresh action.

 And next I change the code to below:
```js

    _onLoadMore(args: LoadOnDemandListViewEventData) {
        const listView: RadListView = args.object;
        if (this._totalCount > -1 && this._totalCount > this._list.length) {
            console.log('on load more');
            args.returnValue = true;
            this.onLoadMore.emit({offset: this._list.length, event: args});
        }
        else {
            console.log('disable load more');
            listView.notifyLoadOnDemandFinished();
            args.returnValue = false;
        }
    }
```
The only difference is that I call a `listView.notifyLoadOnDemandFinished();` to make sure the activityIndicator disappear. And result is like this:
![screenshot1](https://user-images.githubusercontent.com/7570175/44377031-2fef2000-a54f-11e8-9c91-b7692ce4eb80.gif)

console log is this:
```cli
CONSOLE LOG file:///app/vendor.js:137971:31: start waiting
CONSOLE LOG file:///app/bundle.js:6615:24: action:  [orders]load orders
CONSOLE LOG file:///app/bundle.js:6619:32: payload:
{"status":"shipped","forceReload":true,"offset":0}
```

So to put a summary, the issue is:
- Init list  with a short source
- Use `returnValue=false` to disable the listView and `listView.notifyLoadOnDemandFinished();` to disappear the activityIndicator
- perform a pullToRefresh action and the list will gone blank.

I've also update the playground demo. https://play.nativescript.org/?template=play-ng&id=tyV6XG&v=5

I'm aware the version of the RadListView in playground  is 3.5.11. But the behaviour is the same with the `3.6.0`. 
Please reopen this issue.

**Comment by VladimirAmiorkov** [comment]
Hi @gogoout ,

Sorry for the late response. I will take a look at this and get back to you as soon as possible.

**Comment by VladimirAmiorkov** [comment]
Hi @gogoout ,

I was able to reproduce this issue when having a source equal to exactly 1 item anything above causes the `returnValue` to work as expected. I am reopening this issue.

**Comment by gogoout** [comment]
Hi @VladimirAmiorkov ,
I found another bug when I test on different simulator. The recording gif I attached before is run on a Iphone X 11.4 simulator. The one below is running on Iphone 6s 11.4 simulator (also can be reproduced on a Iphone 6s 11.4 real device). 
Please note the code running on both simulators are the same, the source of the list is the same. The only difference between these 2 devices is the screen size.

The actions to trigger the bug is below:
- having a list inited with a source which have almost exactly height as the list have
- perform any slightly scroll movement can cause the list go blank.

Not sure if these 2 are the same bug.
![screenshot6s](https://user-images.githubusercontent.com/7570175/44614364-292a1b00-a877-11e8-991e-d2eedba23557.gif)

**Comment by VladimirAmiorkov** [comment]
Hi @gogoout ,

I would ask you to log this issue in a separate thread to make sure the main issue in this one is not shadowed, also this can allow you to provide full details of this behavior + all of the required details in order to reproduce it on our side.

**Comment by zbranzov** [comment]
Hi @gogoout,
We've published a fix in nativescript-ui-listview@3.7.1. I hope it helps.

**Comment by VladimirAmiorkov** [comment]
@gogoout The difference is that there was a border case where the bound source to the items is smaller than the maximum view port size and the loadOnDemand is set to "Auto" (meaning "load on demand" will be raised immediately during RadListView creation) and at the same time the "load on demand" would never add new items (calling directly disable) would lead to again the "blank list" issue. This case is resolved in the 3.7.1 version

**Comment by gogoout** [comment]
@VladimirAmiorkov emmm.. In my case the situation you mentioned didn't seems to cause the "blank list". Anyway, it's working now, thanks for the fix.
